packages/app/src/systems/Assets/hooks/useAsset.tsx

```diff
@@ -0,0 +1,40 @@
+import { useMemo } from 'react';
+import { isSameEthAddress } from '~/systems/Chains/eth/utils/address';
+
+import type { Asset } from '../services/asset';
+import { getAssetEth, getAssetFuel } from '../utils';
+
+import { useAssets } from './useAssets';
+
+export const useAsset = (params?: {
+  ethTokenId?: string;
+  fuelContractId?: string;
+  fuelTokenId?: string;
+}) => {
+  const { ethTokenId, fuelTokenId, fuelContractId } = params || {};
+  const { assets } = useAssets();
+
+  const asset = useMemo((): Asset | undefined => {
+    // consider ETH as default asset
+    if (!ethTokenId && !fuelTokenId && !fuelContractId) {
+      return assets.find((asset) => asset.symbol === 'ETH');
+    }
+
+    const appAsset = assets.find((asset) => {
+      const ethAsset = getAssetEth(asset);
+      const fuelAsset = getAssetFuel(asset);
+
+      return (
+        isSameEthAddress(ethAsset?.address, ethTokenId) ||
+        (fuelContractId && fuelAsset?.contractId === fuelContractId) ||
```

Comment:
The contractId should not be considered here. As one contract can have multiple asset IDs, what can make the hook to return the wrong asset.

When dealing with asset the only way to identify it is by the assetId or the contractId + subId but never the contractId alone.